pkg/config/config.go

```diff
@@ -79,67 +82,35 @@ func listUserSSHKeys() ([]string, error) {
 // askUserForHostSSHKey asks the user for the SSH key path.
 func askUserForHostSSHKey(keys []string, host *hostcfg) error {
 	if len(keys) == 0 {
-		return survey.AskOne(&survey.Input{Message: "SSH Key path:"}, &host.KeyPath)
+		host.KeyPath = prompts.Input("SSH key path:", "", true)
+		return nil
 	}
 	keys = append(keys, "other")
 	if host.KeyPath == "" {
 		host.KeyPath = keys[0]
 	}
-	if err := survey.AskOne(&survey.Select{
-		Message: "SSH key path:", Options: keys, Default: host.KeyPath,
-	}, &host.KeyPath); err != nil {
-		return fmt.Errorf("unable to ask for ssh key: %w", err)
-	}
+	host.KeyPath = prompts.Select("SSH key path:", keys, host.KeyPath)
 	if host.KeyPath == "other" {
-		return survey.AskOne(&survey.Input{Message: "SSH Key path:"}, &host.KeyPath)
+		host.KeyPath = prompts.Input("SSH key path:", "", true)
 	}
 	return nil
 }
 
 // askUserForHostConfig collects a host SSH configuration interactively.
 func askUserForHostConfig(keys []string, host *hostcfg) error {
 	logrus.Infof("Please provide SSH configuration for the host.")
-	if err := survey.Ask([]*survey.Question{
-		{
-			Name:     "address",
-			Validate: survey.Required,
-			Prompt: &survey.Input{
-				Message: "Node address:",
-				Default: host.Address,
-			},
-		},
-		{
-			Name: "role",
-			Prompt: &survey.Select{
-				Message: "Node role:",
-				Options: []string{"controller+worker", "controller", "worker"},
-				Default: host.Role,
-			},
-		},
-		{
-			Name: "port",
-			Prompt: &survey.Input{
-				Message: "SSH port:",
-				Default: strconv.Itoa(host.Port),
-			},
-			Validate: func(ans interface{}) error {
-				if _, err := strconv.Atoi(ans.(string)); err != nil {
-					return fmt.Errorf("port must be a number")
-				}
-				return nil
-			},
-		},
-		{
-			Name:     "user",
-			Validate: survey.Required,
-			Prompt: &survey.Input{
-				Message: "SSH user:",
-				Default: host.User,
-			},
-		},
-	}, host); err != nil {
-		return fmt.Errorf("unable to ask for host config: %w", err)
+	host.Address = prompts.Input("Node address:", host.Address, true)
+	host.User = prompts.Input("SSH user:", host.User, true)
+	var err error
+	var port int
+	for port == 0 {
+		str := prompts.Input("SSH port:", strconv.Itoa(host.Port), true)
+		if port, err = strconv.Atoi(str); err != nil {
+			logrus.Warnf("Invalid port number")
```

Comment:
just curious why this is a warning and doesnt return an error?

Reply:
We allow users to type the value again if they by mistake entered a string there.